plugins/GSdx/Renderers/Common/GSTextureCache.cpp

```diff
@@ -2116,11 +2116,13 @@ std::size_t GSTextureCache::PaletteKeyHash::operator()(const PaletteKey &key) co
 // GSTextureCache::PaletteKeyEqual
 
 bool GSTextureCache::PaletteKeyEqual::operator()(const PaletteKey &lhs, const PaletteKey &rhs) const {
-	ASSERT(lhs.pal == rhs.pal); // By design, each map SHOULD contain only PaletteKey with the same pal value
-	
-	uint16 pal = lhs.pal;
-	uint16 palette_size = pal * sizeof(uint32);
-	return GSVector4i::compare64(lhs.clut, rhs.clut, palette_size);
+	if (lhs.pal != rhs.pal) {
+		return false;
+	}
+
+	ASSERT((lhs.pal & 15) == 0);
```

Comment:
Why the assert ? It doesn't make any sense here

Reply:
Yeah maybe you are right there is already the ASSERT in the `GSVector4i::compare64`, I can remove it if needed

Reply:
yes because this assert belong to compare64 not this function.